core/vdbe/mod.rs

```diff
@@ -470,7 +471,9 @@ impl Program {
             // invalidate row
             let _ = state.result_row.take();
             let (insn, insn_function) = &self.insns[state.pc as usize];
-            trace_insn(self, state.pc as InsnReference, insn);
+            if enable_tracing {
+                trace_insn(self, state.pc as InsnReference, insn);
+            }
```

Comment:
Isn't there a `#[cfg(tracing_enabled)]` or something like that so we can skip runtime if statement?